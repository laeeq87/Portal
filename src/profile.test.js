import React from "react";
import { render, screen, cleanup} from '@testing-library/react';
import Profile from "./profile";
import { createMemoryHistory } from 'history'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { Router } from "react-router";

afterEach(()=> {
    cleanup();
});

const props = {
    match:{
        params: jest.fn(),
    }
  }

it("Should Render Porfile Component", ()=> {
    const history = createMemoryHistory(['/', '/login']);
    render(
        <Router history={history}>
            <Profile {...props} history={history}/>
        </Router>
    );
    const profileComponentId  = screen.getByTestId('tst-profile')
    expect(profileComponentId).toBeInTheDocument();
}); 


test('matches snapshot', ()=> {
    const history = createMemoryHistory(['/', '/login', '/account']);
    const tree = renderer.create(
        <Router history={history}>
            <Profile {...props} history={history}/>
    </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})
