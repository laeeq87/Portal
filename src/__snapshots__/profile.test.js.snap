// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="tst-profile"
>
  <div
    className="d-flex justify-content-center row row-cols-md-4 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Profile A
          </div>
          <p
            className="card-text"
          >
            Herr Max Mustermann
          </p>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Profile B
          </div>
          <p
            className="card-text"
          >
            Frau Mina Musterfrau
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
